### ISSUE TITLE
100% sql free core installation

### ISSUE THREAD

**Issue by Aokromes** [TrinityCore/TrinityCore#19066]
**Description:**

100% sql free installation

If databases are missing show the following message:

"Databases missing, we need root user password or user with CREATE and GRANT permisions press enter for root user or write user name with CREATE and GRANT permisions"

"we will create 3 (4) databases named auth_335, characters_335, world_335, if you want to use diferent names write them separated by "," don't use spaces on database names, press enter for default names"

"we will create one user to allow to use the databases, by default this user will be trinitycore, press enter to accept or write desired username"

"we will give (user name) one random password, it will be show after this row, press enter to accept or write your desired password"

and after that update conf files

**Current behaviour:** (Tell us what happens.)

you need to run create_mysql.sql, then start core.

**Expected behaviour:** (Tell us what should happen instead.)
you start core.

**Branch(es):** 3.3.5 / master (Tell us which branch(es) this issue affects.)
ALL
**TC rev. hash/commit:**  
0659a85  
**TDB version:**  
ALL
**Operating system:**  
ALL

**Comment by ccrs** [comment]
cant reproduce on 3.3.5

**Comment by Naios** [comment]
Seems like an invalid issue ,-P

**Comment by MrSmite** [comment]
*ALL* is not a version

Kidding aside, I think this is a great idea but please allow it to be optional. I prefer to do my DB stuff manually to keep my mind up to date.

**Comment by tkrokli** [comment]
To me, the issue post is somewhat confusing or misleading (not sure if that is the case for others). The only thing that hinders automatic creation of the databases in a completely new installation is the fact that the local MySQL server is not yet configured to

`GRANT USAGE ON * . * TO 'trinity'@'localhost' IDENTIFIED BY 'trinity' [...]`
+
`GRANT ALL PRIVILEGES ON` `world/characters/auth/hotfixes` `. * TO 'trinity'@'localhost' [...]`

With that fixed, the core is perfectly capable of creating the missing DBs as specified in the .conf files.

I can appreciate the suggestion to write back the names of custom DB names to the .conf files. Myself, I use `335_` as a DB prefix (instead of suffix) for the DBs in my 3.3.5 branch, grouping the DBs in my GUI editor.

**Comment by Aokromes** [comment]
@tkrokli the idea of this all is to fix  reduce the problems for starters AND increase security.

With this system the db section of wiki will be totally deprecated, because the core will do all the needed job of create users and databases and add permissions to users.

We increase security, no longer "trinity" user and password on machines with phpmyadmin allowing hackers to break into databases.

We stop people reusing database names for different wow versions by mistake or bad knowledge and breaking core.

We stop the need of edit sql files / conf files to allow custom database names.

We also can stop the need of "create" conf files, the core can do the job of copy .conf.dist files and populate database names and password without human intervention.

**Comment by Rochet2** [comment]
At the moment its great though how you can have multiple config files, maybe in different folders of cores and just drop the DB to automatically recreate the tables. This would no longer be possible because you would need to drop the tables anyways and you are forced to re enter the table information correctly as its not automatically read or written to configs.

If the core asks for the table names and user passwords etc then it would be great if there was an option in config to use the current system for the table names and user name and passwords taken from config.

**Comment by Aokromes** [comment]
well, some additional questions can be added if we find users/databases names/database content already exists.

**Comment by Naios** [comment]
We could also change the core config into a more machine readable format like `yaml` or `json` and alter the config when doing some core actions so that commands may also affect the permanent config.

**Comment by Aokromes** [comment]
maybe credentials must go on standalone file.

**Comment by Naios** [comment]
Theoretically we could also provide the possibility to split config files in general

**Comment by MrSmite** [comment]
Personally I like the flexibility we have now which allows me to have multiple realms. For example, in your DB you could have:

335_pve_chars
335_pve_world
335_pvp_chars
335_pvp_world
335_test_chars
335_test_world

335_pve.conf
335_pvp.conf
335_test.conf

Each of the DBs could have a different admin, allowing different users to be able to moderate different realms without having access to all of them. From the discussion above it seems as if this would be difficult with the suggested implementation.

**Comment by Aokromes** [comment]
you missed the point "If databases are missing, show the following messages:" :P

**Comment by MrSmite** [comment]
> you missed the point "If databases are missing, show the following messages:" :P

No, I saw that. I was just saying it seems like trying to solve a problem that doesn't need solving. I haven't seen very many people say it's too hard to get started.

**Comment by Aokromes** [comment]
if you create the databases (with the "create_mysql or hand create) and edit config, the system will skip points 2 to 4. And yes, it's a problem that needs to be solved because to much people is having problems.

**Comment by DDuarte** [comment]
> Current behaviour: (Tell us what happens.) You need to run create_mysql.sql, then start core.

As long as you configure the host, username and password correctly in worldserver.conf, you do not need create_mysql.sql or to create databases manually, worldserver already does that for you.

**Comment by Aokromes** [comment]
if you setup the databases to use existing users/use root has credentials.

**Comment by Seyden** [comment]
it happens so often that a certain sql update is missing and i get angry that i can't test my stuff, because a field is missing. Thats when i would love the idea that the core just auto creates all databases.. in that case i don't have to case anymore about inserting all sqls

**Comment by Naios** [comment]
What we could do is to ask whether the core should be started anyway in case of an update fails to apply.

**Comment by Aokromes** [comment]
No, better no, that can end on strange crashes.

**Comment by MrSmite** [comment]
> With this system the db section of wiki will be totally deprecated

I still like the overall idea but I disagree here. The wiki is still useful for people who want to add custom stuff, also, there are times when the wiki has helped to make some of the internal code easier to understand.

**Comment by Aokromes** [comment]
ofc installation only :)

**Comment by Aokromes** [comment]
added 1 thing more to make installation more idiot-prof safe:

If databases are present, check they contain all needed tables.

If they don't contain needed tables warn about database(s) is used by some other project and don't allow to use then to install tc.

**Comment by Rushor** [comment]
@MrSmite are you still here?

**Comment by MrSmite** [comment]
> @MrSmite are you still here?

Yes, occasionally

**Comment by Rushor** [comment]
Nice man :)

**Comment by Aokromes** [comment]
maybe even if TDB file is missing download from github.

**Comment by Aokromes** [comment]
still wanted.
